src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/ColorPaletteIcon/index.js

```diff
@@ -8,13 +8,25 @@ const ColorPaletteIcon = ( {
 	setSelectedColor,
 	colors,
 	setShowCustomColors = null,
+	customColors,
 } ) => {
-	const conicGradient = `conic-gradient(${ colors[ idx ].primary } 90deg, ${ colors[ idx ].secondary } 90deg 150deg, ${ colors[ idx ].tertiary } 150deg 330deg, ${ colors[ idx ].primary } 330deg 360deg)`;
+	let conicGradient;
+	if ( colors ) {
+		conicGradient = `conic-gradient(${ colors[ idx ].primary } 90deg, ${ colors[ idx ].secondary } 90deg 150deg, ${ colors[ idx ].tertiary } 150deg 330deg, ${ colors[ idx ].primary } 330deg 360deg)`;
+	} else if ( customColors ) {
+		conicGradient = `conic-gradient(${ customColors.primary } 90deg, ${ customColors.secondary } 90deg 150deg, ${ customColors.tertiary } 150deg 330deg, ${ customColors.primary } 330deg 360deg)`;
+	}
+
```

Comment:
Can we pass a single data structure to the component and use it everywhere? The prop `colors` can be either `colors[idx]` or `customColors`.